cmd/mp2ts-info/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/Eyevinn/mp2ts-tools/common"
+	"github.com/asticode/go-astits"
+)
+
+var usg = `Usage of %s:
+
+%s lists information about TS files, e.g. pids, bitrates, service, etc
+`
+
+func parseOptions() common.Options {
+	opts := common.Options{ShowStreamInfo: true}
+	flag.BoolVar(&opts.ShowService, "service", false, "show service information")
+	flag.BoolVar(&opts.Indent, "indent", false, "indent JSON output")
+	flag.BoolVar(&opts.Version, "version", false, "print version")
+
+	flag.Usage = func() {
+		parts := strings.Split(os.Args[0], "/")
+		name := parts[len(parts)-1]
+		fmt.Fprintf(os.Stderr, usg, name, name)
+		fmt.Fprintf(os.Stderr, "\nRun as: %s [options] file.ts (- for stdin) with options:\n\n", name)
+		flag.PrintDefaults()
+	}
+
+	flag.Parse()
+	return opts
+}
+
+func ParseInfo(ctx context.Context, w io.Writer, f io.Reader, o common.Options) error {
```

Comment:
This is a pretty long function, so it should be tested somehow. 
There is also no reason to have it public since main functions cannot be exported.